### PR TITLE
Raise error when obsolete argument is given

### PR DESCRIPTION
**Pull request by unnonouno**
I added obsolete argument to show error.

### CODE REVIEW

**Comment by delta2323** [comment]
I would like to confirm my understanding. If users use `BatchNormalization.__call__` with non-default `test` and default `finetune` argument (i.e. `bn(x, t, False)`) in v1, users hardly realize that its API is changed because it does not raise any error, although the third argument is interpreted as `finetune` not `test`. The objective is to notify users in such a case.

**Comment by unnonouno** [comment]
I finally decided to use `kwargs` for both `test` and `finetune`. I mean positional arguments are inhibited. Users always need to use keyword arguments. If they use a positional argument, simply `TypeError` is raised.

**Comment by beam2d** [comment]
Can you make the check of `test` argument as an independent method? We have to add this argument check to many functions and this checking can be abstracted by a method/decorator. It may look like follows:

```python
def __call__(self, **kwargs):
    finetune = kwargs.pop('finetune', False)
    check_unexpected_kwargs(
        kwargs, test='test argument is removed in Chainer v2. Use chainer.config.train instead.')
```

The `check_unexpected_kwargs` checks if no argument is left and it emits a special message for specified ones (in this case, `test`).

**Comment by unnonouno** [comment]
It sounds good. And we can provide another helper method:
```
finetune, = parse_kwargs(['finetune'], kwargs)
```

File: chainer/links/normalization/batch_normalization.py [hunk 1]
```diff
@@ -84,13 +85,20 @@ def __init__(self, size, decay=0.9, eps=2e-5, dtype=numpy.float32,
...
+
+        Invokes the forward propagation of BatchNormalization.
 
         In training mode, the BatchNormalization computes moving averages of
         mean and variance for evaluatino during training, and normalizes the
         input using batch statistics.
 
+        .. warning::
+
+           Do not use ``test`` argument. This argument is obsolete in v2.
```
Comment (beam2d) [inline chainer/links/normalization/batch_normalization.py]: `test` is actually not just made obsolete but not supported anymore. How about changing this line to
```
``test`` argument is not supported anymore since v2...
```

File: chainer/links/normalization/batch_normalization.py [hunk 2]
```diff
@@ -84,13 +85,20 @@ def __init__(self, size, decay=0.9, eps=2e-5, dtype=numpy.float32,
...
+           Instead use ``chainer.using_config('train', train)``.
```
Comment (beam2d) [inline chainer/links/normalization/batch_normalization.py]: Instead, (add comma)

File: chainer/links/normalization/batch_normalization.py [hunk 3]
```diff
@@ -100,6 +108,10 @@ def __call__(self, x, finetune=False):
                 statistics.
 
         """
+        argument.check_unexpected_kwargs(
+            kwargs, test='test argument is obsolete. Use chainer.using_config')
```
Comment (beam2d) [inline chainer/links/normalization/batch_normalization.py]: Here again, how about changing `obsolete` to `not supported anymore`?

File: chainer/utils/argument.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+        if key in kwargs:
+            raise ValueError(message)
+
+
+def parse_kwargs(kwargs, *name_and_values):
+    values = [kwargs.pop(name, default_value)
+              for name, default_value in name_and_values]
+    if kwargs:
+        args = ', '.join(["'%s'" % arg for arg in kwargs.keys()])
+        raise TypeError('got an unexpected keyword argument %s' % args)
```
Comment (beam2d) [inline chainer/utils/argument.py]: `an unexpected keyword argument` -> `unexpected keyword argument(s)` (or write more to change the wording for a single arg and multiple args)

**Comment by beam2d** [comment]
LGTM except for another comment.
